src/Hazelcast.Net/Sql/HazelcastSqlException.cs

```diff
@@ -13,26 +13,117 @@
 // limitations under the License.
 
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
 using Hazelcast.Exceptions;
+using Hazelcast.Core;
 
 namespace Hazelcast.Sql
 {
+    /// <summary>
+    /// Represents the exception that is thrown by the SQL service in case of an error.
+    /// </summary>
+    [Serializable]
```

Comment:
Just a question: What does this annotation(?), or whatever it is called, do?

Reply:
Such an annotation is called an "attribute" marking the class, in C#. It indicates the framework that the class can be serialized via the built-in binary serializer, if needed. It's usually a good idea to avoid this, but is recommended for exceptions.